frontend/src/components/Link/KialiPageLink.tsx

```diff
@@ -2,38 +2,61 @@ import * as React from 'react';
 import { Link } from 'react-router-dom';
 import { ExternalLinkAltIcon } from '@patternfly/react-icons';
 import { serverConfig } from '../../config';
+import {KialiAppState} from "../../store/Store";
+import {connect} from "react-redux";
+import {kioskContextMenuAction} from "../Kiosk/KioskActions";
 
-interface KialiPageLinkProps {
+type ReduxProps = {
+    kiosk: string;
+}
+
+type KialiPageLinkProps = ReduxProps & {
+  children: React.ReactNode;
   cluster?: string;
   href: string;
 }
 
-const KialiPageLink: React.FC<KialiPageLinkProps> = props => {
-  // Without a cluster, simply render a local link
-  if (props.cluster === undefined) {
-    return <Link to={props.href}>{props.children}</Link>;
-  }
+class KialiPageLink extends React.Component<KialiPageLinkProps> {
 
-  // If cluster is specified, and it's the home cluster, render a local link.
-  if (!serverConfig.clusterInfo?.name || props.cluster === serverConfig.clusterInfo.name) {
-    return <Link to={props.href}>{props.children}</Link>;
-  }
+  render() {
+    // Without a cluster, simply render a local link
+    // If cluster is specified, and it's the home cluster, render a local link.
+    if (this.props.cluster === undefined || !serverConfig.clusterInfo?.name || this.props.cluster === serverConfig.clusterInfo.name) {
+      if (this.props.kiosk.length > 0 && this.props.kiosk !== 'true') {
```

Comment:
This test is becoming so frequent that perhaps a simple utility would be nice, `isParentKiosk(this.props.kiosk)`.

Reply:
It's a good one, but I'm reserving this change for follow-up refactoring.
I think we need to keep the kiosk parameter for "true/false" values and also introduce another set of kiosk aren't, kioskXYZ for additional parameters that it may need.

The next tasks is to work in the links from the "Service Mesh" tab to Kiali, and in that context, I'd apply this recommendation.